src/d3d9/d3d9_device.cpp

```diff
@@ -6297,6 +6313,64 @@ namespace dxvk {
   }
 
 
+  void D3D9DeviceEx::UpdateTextureTypeMismatchesForShader(const D3D9CommonShader* shader, uint32_t shaderSamplerMask, uint32_t shaderSamplerOffset) {
+    if (unlikely(shader->GetInfo().majorVersion() < 2 || m_d3d9Options.forceSamplerTypeSpecConstants)) {
```

Comment:
Why are you unlikely-ing the entire d3d8 frontend? :stuck_out_tongue: I'd just not hint this bit at all.

Reply:
Because old games are less unlikely to be demanding, so I'd rather optimize for the newer ones, especially when we're talking about tiny things like the `likely/unlikely` hints.

Reply:
There are plenty of SM1 d3d9 games out there too, but fine, have it your way.